src/debug-generator/debug-generator.c

```diff
@@ -64,12 +65,29 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
                 const char *t = NULL;
 
                 r = value ? parse_boolean(value) : 1;
+                if (r == 0) {
+                        arg_debug_shell = false;
+                        return 0;
+                }
+
+                arg_debug_shell = true;
+
                 if (r < 0)
                         t = skip_dev_prefix(value);
-                else if (r > 0)
+                else if (r > 0) {
+                        if (arg_debug_tty)
+                                return 0;
+
                         t = skip_dev_prefix(DEBUGTTY);
+                }
+
+                return free_and_strdup_warn(&arg_debug_tty, t);
+
+        } else if (proc_cmdline_key_streq(key, "systemd.default_debug_tty")) {
+                if (proc_cmdline_value_missing(key, value))
+                        return 0;
 
-                return free_and_strdup_warn(&arg_debug_shell, t);
+                return free_and_strdup_warn(&arg_debug_tty, skip_dev_prefix(value));
```

Comment:
this is a bit weird, because it means that a cmdline such as `systemd.debug_tty=tty7 systemd.default_debug_tty=tty8` would result in a the debug shell on tty8, but it really should be on tty7...

I am pretty sure you should settle the tty to use at time of use, not at time of parse. this will otherwise just get confusing...